code/EventLinkQR/app/src/main/java/com/example/eventlinkqr/LandingPage.java

```diff
@@ -35,5 +42,31 @@ public void onClick(View v) {
             startActivity(intent);
             finish(); // Close the landing page activity
         }
+
+        FirebaseMessaging.getInstance().getToken()
+                .addOnCompleteListener(task -> {
+                    if (!task.isSuccessful()) {
+                        Log.w(TAG, "Fetching FCM registration token failed", task.getException());
+                        return;
+                    }
+                    // Get new FCM registration token
+                    String token = task.getResult();
+                    // Log and toast
+                    Log.d(TAG, "FCM Token: " + token);
+                });
+
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
Not sure if we can do this. We might need to ask Akalanka to see if they will test it on an old device.

Reply:
Yes, it is necessary when dealing with functionalities that are only available in Android Oreo (API level 26) and above. The if condition is specifically for Android Oreo and newer versions because they require notification channels for notifications. For devices older than Oreo, the notification will still work without needing a channel. I apologize because I should have this as a comment in the code.